components/canvas/webgl_thread.rs

```diff
@@ -18,6 +18,33 @@ use webrender_api;
 /// It allows to get a WebGLThread handle for each script pipeline.
 pub use ::webgl_mode::WebGLThreads;
 
+struct GLContextData {
+    ctx: GLContextWrapper,
+    state: GLState,
+}
+
+pub struct GLState {
+    clear_color: (f32, f32, f32, f32),
+    scissor_test_enabled: bool,
+    stencil_write_mask: (u32, u32),
+    stencil_clear_value: i32,
+    depth_write_mask: bool,
+    depth_clear_value: f64,
+}
```

Comment:
Why is this on the GL side when it could be on the DOM side?

Reply:
I'm strongly against caching *some* stuff on the GL side and *some* stuff on the DOM side. All of the bookkeeping I've implemented this year has been put on the DOM side to be able, I see no reason to put those values on the GL side.

Reply:
Never mind I see why now.